webui/react/src/pages/ExperimentList.tsx

```diff
@@ -754,31 +756,15 @@ const ExperimentList: React.FC<Props> = ({ project }) => {
     [fetchExperiments, sendBatchActions, updateSettings],
   );
 
-  const showConfirmation = useCallback(
-    (action: Action) => {
-      modal.confirm({
-        content: `
-        Are you sure you want to ${action.toLocaleLowerCase()}
-        all the eligible selected experiments?
-      `,
-        icon: <ExclamationCircleOutlined />,
-        okText: /cancel/i.test(action) ? 'Confirm' : action,
-        onOk: () => submitBatchAction(action),
-        title: 'Confirm Batch Action',
-      });
-    },
-    [submitBatchAction],
-  );
-
   const handleBatchAction = useCallback(
     (action?: string) => {
       if (action === Action.OpenTensorBoard || action === Action.Move) {
         submitBatchAction(action);
-      } else {
-        showConfirmation(action as Action);
+      } else if (action) {
+        setBatchAction(action as Action);
```

Comment:
call `open()` here

Reply:
> Previous review/comment was based on the assumption that `ConfirmModal` was only being used in `ExperimentList`. But I see that similar confirmation modals still also need to be converted in `PaginatedNotesCard`, `TaskList`, and `ActionDropdown`. So `ExperimentBatchActionModal` may be too specific.
> 
> I think something like this would address all of these cases (and potentially `TaskKillModal`, `ExperimentDeleteModal`, `ExperimentStopModal`, and all the other similar confirmation modals that confirm a Delete action):
> 
> `<ConfirmModal.Component batch={boolean} action={actionName} onConfirm={handler}/>`
> 
> The copy for these is consistent enough that we can just set it in `ConfirmModal` and switch out the action name or make it singular/plural. Can probably just use the `danger` icon for all of them.
> 
> The copy for the `PaginatedNotesCard` modal is perhaps different enough to justify its own modal. Or maybe we can just add a prop to overwrite the copy in particular cases.
> 
> I guess my overall point here is just that I think the `ConfirmModal`, to be as useful as possible, should take descriptive props and generate props for an instance of `Modal`, instead of just passing along props that are basically the same ones that `Modal` uses.

This is a great review. It seems like there are two classes of confirmation modal uses. One for batch actions and another for a more generalized confirm modal. I think a more specialized batch action confirm modal is warranted with props that are more geared specific to batch action confirmation. As for the general confirmation, besides the current `ConfirmModal` setup where there is a dedicated modal component for confirmation modals, is there a different setup that would be better that I'm not thinking about? The only alternative I could think of was to define a custom `Modal` within the respective place where a confirmation modal is needed, and that approach seems repetitive.